### PR TITLE
IA-2847 Cache latest AoU and GATK image

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadworkbench.atlassian.net/browse/IA-2847

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [ ] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: jenkins/dataproc-custom-images/prepare-custom-leonardo-jupyter-dataproc-image.sh [hunk 1]
```diff
@@ -22,9 +22,9 @@ terra_jupyter_python="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-python:1.0.0"
 terra_jupyter_r="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-r:2.0.0"
 terra_jupyter_bioconductor="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-bioconductor:2.0.0"
 terra_jupyter_hail="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-hail:1.0.0"
-terra_jupyter_gatk="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-gatk:2.0.0"
-terra_jupyter_aou_old="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-aou:1.1.6"
-terra_jupyter_aou="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-aou:1.1.6"
+terra_jupyter_gatk="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-gatk:2.0.1"
+terra_jupyter_aou_old="us.gcr.io/broad-dsp-gcr-public/terra-jupyter-aou:1.1.2"
```
Comment (kyuksel) [inline jenkins/dataproc-custom-images/prepare-custom-leonardo-jupyter-dataproc-image.sh]: I have a couple of questions:

1. Why are we lowering the version of `aou_old`?
2. AoU is using [terra-jupyter-aou:1.1.5](https://github.com/all-of-us/workbench/blob/ba26a89ea09c21e29cec73113e683b44641774de/api/config/config_prod.json#L13). Shouldn't we have been caching that?

Comment (Qi77Qi) [reply jenkins/dataproc-custom-images/prepare-custom-leonardo-jupyter-dataproc-image.sh]: when I checked, aou was still using 1.1.2...https://terra-bio.slack.com/archives/C9C4KKU65/p1626958772093300

Comment (als364) [reply jenkins/dataproc-custom-images/prepare-custom-leonardo-jupyter-dataproc-image.sh]: AoU is [currently on 1.1.5](https://github.com/all-of-us/workbench/blob/master/api/config/config_prod.json#L13).

File: README.md [hunk 2]
```diff
@@ -49,7 +49,7 @@ Run `docker system prune -a`
 
 Build Leonardo and run all unit tests.
 ```
-export SBT_OPTS="-Xmx2G -Xms1G -Dmysql.host=localhost -Dmysql.port=3311 -Duser.timezone=UTC"
+export JAVA_OPTS="-Dheadless=false -Duser.timezone=UTC"
```
Comment (kyuksel) [inline README.md:52]: why is this change?

File: http/src/main/resources/reference.conf [hunk 3]
```diff
@@ -61,7 +61,7 @@ dataproc {
     numberOfPreemptibleWorkers = 0
     region = "us-central1"
   }
-  customDataprocImage = "projects/broad-dsp-gcr-public/global/images/test-1-4-51-debian10-qi-2021-07-21-16-54-07"
+  customDataprocImage = "projects/broad-dsp-gcr-public/global/images/test-1-4-51-debian10-qi-2021-07-22-11-27-42"
```
Comment (kyuksel) [inline http/src/main/resources/reference.conf:64]: Why is the naming scheme different? They used to start with `leonardo-custom-image` or something like that.

File: README.md (hunk 2 above)
Comment (Qi77Qi) [reply README.md:52]: I don't think `SBT_OPTS` works anymore..probably due to sbt version upgrade

File: http/src/main/resources/reference.conf (hunk 3 above)
Comment (Qi77Qi) [reply http/src/main/resources/reference.conf:64]: these images are still generated locally..I didn't really bother to find out what string is used in jenkins

Comment (kyuksel) [reply http/src/main/resources/reference.conf:64]: We could still name them the same way as before. It's sometimes helpful to be able to filter for images with a known simple prefix.

**Review by kyuksel — APPROVED** [review]

File: http/src/main/resources/reference.conf (hunk 3 above)
Comment (Qi77Qi) [reply http/src/main/resources/reference.conf:64]: hmm..I've seen various prefixes or postfixes..but yea, I can I see a consistent prefix can be helpful...altho originally I wanted to differentiate locally generated images from jenkins generated ones

Comment (rtitle) [reply http/src/main/resources/reference.conf:64]: Is the jenkins job still not working?

**Review by rtitle — APPROVED** [review]

File: http/src/main/resources/reference.conf (hunk 3 above)
Comment (Qi77Qi) [reply http/src/main/resources/reference.conf:64]: nope...but it seems things are better after merging the previous PR..see https://fc-jenkins.dsp-techops.broadinstitute.org/view/Leonardo/job/leonardo-custom-image/194/

**Pull request merged**